### ISSUE TITLE
Data duplicated in multi GPU evaluation

### ISSUE THREAD

**Issue by Fengdalu** [huggingface/accelerate#125]
Data may be duplicated when evaluation using multi GPUs, 

For example, batch_size = 320 per GPU, and the total number of evaluation samples is 25000, When using 4 GPUs, the final data count would be 25600 (4 * 20 * 320), even set `split_batches=True`.

Setting accurate batch size can divide by the total number of data can solve it. Besides, how to load each data once in evaluation?

@sgugger

**Comment by sgugger** [comment]
Yes, you need a round multiple of the number of processes and accelerators like TPUs need to have fixed shapes, so the dataloarder go back over the data at the beginning to fill the last batch if you don't specify `drop_last=True`. Note that for training you go through your training data several times, and as long as you have `shuffle=True` in your training dataloader, those duplicate examples will be different at each epoch.

For evaluation, you can just truncate the arrays of predictions to the length of your validation dataset, to ignore those added results.

**Comment by Fengdalu** [comment]
Many Thanks
